src/write.c

```diff
@@ -941,6 +941,30 @@ static avifCodecType avifEncoderGetCodecType(const avifEncoder * encoder)
     return avifCodecTypeFromChoice(encoder->codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
 }
 
+static avifBool avifEncoderDataForceKeyframeForAlpha(const avifEncoderData * data, const avifEncoderItem * colorItem, avifAddImageFlags addImageFlags)
+{
+    if (!data->alphaPresent) {
+        // There is no alpha plane.
+        return AVIF_FALSE;
+    }
+    if (addImageFlags & AVIF_ADD_IMAGE_FLAG_SINGLE) {
+        // Not an animated image.
+        return AVIF_FALSE;
+    }
+    if (data->frames.count == 0) {
+        // This is the first frame, no need to force keyframe.
+        return AVIF_FALSE;
+    }
+    const uint32_t sampleCount = colorItem->encodeOutput->samples.count;
+    avifBool isLaggedOutput = (sampleCount == 0) || (data->frames.count != sampleCount - 1);
```

Comment:
Nit: The following should be equivalent:

```
    avifBool isLaggedOutput = (data->frames.count + 1 != sampleCount);
```

Reply:
Done.